cpp/neuralnet/nninputs.cpp

```diff
@@ -641,6 +641,40 @@ Board SymmetryHelpers::getSymBoard(const Board& board, int symmetry) {
   return symBoard;
 }
 
+void SymmetryHelpers::maskSymmetricDuplicativeLoc(const Board &board, bool *const isSymDupLoc) {
+  fill(isSymDupLoc, isSymDupLoc + Board::MAX_ARR_SIZE, false);
+  vector<int> symTypes;
+  symTypes.reserve(NNInputs::NUM_SYMMETRY_COMBINATIONS-1);
+
+  for (int symmetry = 1; symmetry < NNInputs::NUM_SYMMETRY_COMBINATIONS; symmetry++) {
+    bool isBoardSym = true;
+    for (int y = 0; y < board.y_size; y++) {
+      for (int x = 0; x < board.x_size; x++) {
+        Loc loc = getSymLoc(x, y, board,symmetry);
```

Comment:
Potential bug: If the board x_size and y_size are very different then for symmetries that involve a transpose, you will probably be accessing somewhere very far outside of the boundaries of the board.

Probably should explicitly check and disallow symmetries that involve transpose if the board is not square.

Reply:
Good point and I agreed. Just changed.